src/DSManager.cpp

```diff
@@ -405,15 +419,163 @@ Subscriber* DSManager::removeSubscriber(
+        ReturnCode_t ret;
+
+        for (const auto& writer: writers)
+        {
+            ret = entity.publisher->delete_datawriter(writer);
+            if (ReturnCode_t::RETCODE_OK != ret)
+            {
+                LOG_ERROR("Error deleting Data Writer");
+            }
+            auto topic = writer->get_topic();
+
+            entity.participant->unregister_type(topic->get_type_name());
+
+        }
+        for (const auto& reader: readers)
+        {
+            //auto listener = reader->get_listener();
+            ret = entity.subscriber->delete_datareader(reader);
+            if (ReturnCode_t::RETCODE_OK != ret)
+            {
+                LOG_ERROR("Error deleting Data Reader");
+            }
+
+            auto topic = reader->get_topicdescription();
+
+            entity.participant->unregister_type(topic->get_type_name());
+
+        }
+
+        // Now we can delete the Publishers/Subscribers
+
+        ret = entity.participant->delete_publisher(entity.publisher);
+        if (ReturnCode_t::RETCODE_OK != ret)
+        {
+            LOG_ERROR("Error deleting Publisher");
+        }
+        ret = entity.participant->delete_subscriber(entity.subscriber);
+        if (ReturnCode_t::RETCODE_OK != ret)
+        {
+            LOG_ERROR("Error deleting Subscriber");
+        }
+
+
+        // Now we delete the associated participant
+
+
+        entity.participant->set_listener(nullptr);
+
+
+        for (auto topic: entity.registeredTopics )
+        {
+            entity.participant->delete_topic(topic.second);
+            if (ReturnCode_t::RETCODE_OK != ret)
+            {
+                LOG_ERROR("Error deleting Topic");
+            }
+        }
+
+        entity_map.erase(participant->guid());
+
+    }
+
+    return DomainParticipantFactory::get_instance()->delete_participant(participant);
+
+}
+
+template<>
+void DSManager::setDomainEntityTopic(
+        fastdds::dds::DataWriter* entity,
+        Topic* t)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+
+    entity_map[entity->get_publisher()->get_participant()->guid()].publisherTopic = t;
+}
+
+template<>
+void DSManager::setDomainEntityTopic(
+        fastdds::dds::DataReader* entity,
+        Topic* t)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+
+    entity_map[entity->get_subscriber()->get_participant()->guid()].subscriberTopic = t;
+}
+
+template<>
+void DSManager::setDomainEntityType(
+        fastdds::dds::Publisher* entity,
+        TopicDataType* t)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+
+    entity_map[entity->get_participant()->guid()].publisherType = t;
+}
+
+template<>
+void DSManager::setDomainEntityType(
+        fastdds::dds::Subscriber* entity,
+        TopicDataType* t)
+{
+    std::lock_guard<std::recursive_mutex> lock(management_mutex);
+
+    entity_map[entity->get_participant()->guid()].subscriberType = t;
+}
+
 void DSManager::addPublisher(
```

Comment:
```suggestion
void DSManager::addDataWriter(
```

This suggestion applies to other methods below.